include/boost/gil/extension/numeric/affine.hpp

```diff
@@ -89,6 +89,25 @@ point<F> transform(matrix3x2<F> const& mat, point<F2> const& src)
     return src * mat;
 }
 
+/// Returns the inverse of the given affine transformation matrix
+///
+/// \warning Floating point arithmetic, use Boost.Rational if precision maters
+template <typename T>
+boost::gil::matrix3x2<T> inverse(boost::gil::matrix3x2<T> m)
+{
+    T determinant = m.a * m.d - m.b * m.c;
```

Comment:
Nitpick: I'd generally stick `T const determinant`, especially in bodies of longer functions.